validationTemplate/src/validationWorkflowGenerator.py

```diff
@@ -0,0 +1,414 @@
+        assert isinstance(validationFile, str)
+        assert isinstance(calibration, bool)
+    except AssertionError as e:
+        logger.exception('A test failed : ')
+        raise e
+    # Load template
+    logger.info('Loading validation template.')
+    validation_template = ValidationTemplate(authorized['metrics'], authorized['distributions'], calibration)
+    if calibration and romTag: 
+        validation_template.loadTemplate('validation_ROM_incomplete_XML.xml', os.path.dirname(inspect.getfile(ValidationTemplate)))
+    elif calibration:
+        validation_template.loadTemplate('validation_incomplete_XML.xml', os.path.dirname(inspect.getfile(ValidationTemplate)))
+    else: 
+        validation_template.loadTemplate('validation_incomplete_XML_valonly.xml', os.path.dirname(inspect.getfile(ValidationTemplate)))
+    logger.info('Validation template loaded.')
+
+    # Writing the validation MXL file without running it with RAVEN and get errors
+    logger.info('Writing RAVEN validation input file.')
+    valXMLToWrite = validation_template.createWorkflow(model=model, exp=exp, variables=variables, metric=metric, romTag=romTag)
+    errors = validation_template.writeWorkflow(valXMLToWrite, validationFile, run=False)
+    if errors == 0:
+        logger.info('Successfully wrote input "{}".'.format(validationFile))
+
+
+def writeXMLCalibration(model, exp, variables, metric, optimization, work_dir, calibrationFile, validationFile): 
+    """ 
+    Write XML Calibration input file 
+        @ In, model, dict, model information, executable, SRQ, initialization file 
+        @ In, exp, dict, experimental dataset path, SRQ names, post-processing tag and script 
+        @ In, variables, dict, variables model path, distribution 
+        @ In, metric, string, metric for exp/model comparison
+        @ In, optimization, dict, optimization algorithm information 
+        @ In, work_dir, string, name of working directory to store calibration files 
+        @ In, calibrationFile, string, name of XML RAVEN calibration input file 
+        @ in, validationFile, string, name of XML RAVEN validation input file 
+        @ Out, None
+    """
+    # Check inputs 
+    try:
+        assert isinstance(model, dict)
+        assert isinstance(exp, dict)
+        assert isinstance(variables, dict)
+        assert isinstance(metric, str)
+        assert metric in authorized['metrics'].keys()
+        assert isinstance(optimization, dict)
+        assert isinstance(work_dir, str)
+        assert isinstance(calibrationFile, str)
+        assert isinstance(validationFile, str)
+    except AssertionError as e:
+        logger.exception('A test failed : ')
+        raise e
+    # Load Template 
+    logger.info('Loading calibration template.')
+    calibration_template = CalibrationTemplate(authorized['distributions'], authorized['metrics_sync'])
+    #calibration_template.loadTemplate('calibration_incomplete_XML.xml', os.path.dirname(__file__))
```

Comment:
is this commented code being saved? If so, it would be good to add a comment as to why